migration-engine/core/src/commands/list_migration_directories.rs

```diff
@@ -28,10 +28,13 @@ impl<'a> MigrationCommand for ListMigrationDirectoriesCommand {
     type Input = ListMigrationDirectoriesInput;
     type Output = ListMigrationDirectoriesOutput;
 
-    async fn execute<C: MigrationConnector>(
-        input: &Self::Input,
-        _engine: &MigrationApi<C>,
-    ) -> CoreResult<Self::Output> {
+    async fn execute<C: MigrationConnector>(input: &Self::Input, engine: &MigrationApi<C>) -> CoreResult<Self::Output> {
+        //Validate Provider
+        migration_connector::error_on_changed_provider(
+            &input.migrations_directory_path,
+            engine.connector().connector_type(),
+        )?;
```

Comment:
I am wondering whether we should fail here (the command doesn't have any relation with the schema), but I also think it's fine if we do.